workflow-templates/inginious-task-tester-container-builder.py

```diff
@@ -0,0 +1,17 @@
+#! /bin/env python3
+
+import subprocess
+import os
+
+containers = os.environ.get('REQUIRED', 'base').split(' ')
+container_names = []
+for container in containers:
```

Comment:
Here containers are iteratively built based on what the environment file contains. A choice has to made be made, either to:
- document and tell the users to specify the environments in order such that dependencies are met, otherwise it might behave strangely, crashing or pulling older versions on DockerHub.
- introspect dockerfiles recursively as it is done with `inginious-install` to solve dependencies automatically.

I'm OK with choosing the first option. The second can also be implemented later, as this is not blocking.

In the second case it would be nice to refactor the base code and allow to use it outside of `inginious-install`. This way, in case we add support for external Dockerfiles some day, those scripts would be ready too.

Reply:
As we discussed some time ago, the best solution would be building and pushing INGInious containers automatically on a registry (e.g. DockerHub and/or [GitHub](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry).). We could then simply pull the images from there rather than rebuilding them when the cache is not filled. This would also ensure uniformity between all the CI instances.

I will open a Feature Request on the main repo for that.

Reply:
Should I consider this as a choice for option 1 and that the PR is ready for merge then ?